### PR TITLE
Annotate System.IO.Pipes.AccessControl for nullable reference types

### PR DESCRIPTION
**Pull request by eerhardt**
Contributes to #2339

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -17,6 +17,7 @@
     <add key="dotnet5-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5-transport/nuget/v3/index.json" />
     <add key="dotnet-core" value="https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json" />
     <add key="dotnet-coreclr" value="https://dotnetfeed.blob.core.windows.net/dotnet-coreclr/index.json" />
+    <add key="eerhardt-test" value="https://www.myget.org/F/eerhardt-testfeed/api/v3/index.json" />
```
Comment (eiriktsarpalis) [inline NuGet.config]: this should probably be removed.

File: eng/Versions.props [hunk 2]
```diff
@@ -62,7 +62,7 @@
     <MicrosoftDotNetApiCompatVersion>5.0.0-beta.20112.7</MicrosoftDotNetApiCompatVersion>
     <MicrosoftDotNetBuildTasksFeedVersion>5.0.0-beta.20112.7</MicrosoftDotNetBuildTasksFeedVersion>
     <MicrosoftDotNetCodeAnalysisVersion>5.0.0-beta.20112.7</MicrosoftDotNetCodeAnalysisVersion>
-    <MicrosoftDotNetGenAPIVersion>5.0.0-beta.20112.7</MicrosoftDotNetGenAPIVersion>
+    <MicrosoftDotNetGenAPIVersion>5.0.0-dev-1</MicrosoftDotNetGenAPIVersion>
```
Comment (eiriktsarpalis) [inline eng/Versions.props]: Same here.

File: NuGet.config (hunk 1 above)
Comment (eerhardt) [reply NuGet.config]: Yep. These are temporarily necessary to get the fix for https://github.com/dotnet/arcade/pull/4852. Without it the code won’t build. I’ll revert that temporary commit before merging.

Comment (stephentoub) [reply NuGet.config]: Can you instead just set the nullable context to annotations for the builds in question?

**Review by stephentoub — COMMENTED** [review]
The .cs changes LGTM.

File: NuGet.config (hunk 1 above)
Comment (eerhardt) [reply NuGet.config]: The arcade change has now been merged into dotnet/runtime master. I have reverted the temporary commit to use the local arcade.  There is no need for a workaround.

Comment (stephentoub) [reply NuGet.config]: > There is no need for a workaround.

Can the other workaround cited be removed as well then? Or was that a different issue?

Comment (eerhardt) [reply NuGet.config]: What other workaround cited?

There was an issue with how we generate `.notsupported.cs` files w.r.t. nullable annotations. See https://github.com/dotnet/arcade/pull/4852. I was using a locally built arcade that contained the fix in order to test my changes here. Now that the arcade change has flown into dotnet/runtime, I have removed all workarounds.

Comment (stephentoub) [reply NuGet.config]: The example I cited above in https://github.com/dotnet/runtime/pull/32343#discussion_r379707979 from System.Security.Principal.Windows.

Comment (eerhardt) [reply NuGet.config]: Ah, I missed that was a different project. I thought it was for this project. Yes, I can remove the System.Security.Principal.Windows workaround.

Comment (stephentoub) [reply NuGet.config]: Thanks.

Comment (eerhardt) [reply NuGet.config]: It turns out the workaround in System.Security.Principal.Windows is a different issue - https://github.com/dotnet/arcade/issues/4722. 

https://github.com/dotnet/runtime/blob/6ce761095b57ed38242c8140bc4f1d6da7251567/src/libraries/System.Security.Principal.Windows/ref/System.Security.Principal.Windows.cs#L268

I have updated the comment to point to the underlying arcade issue.

**Review by stephentoub — APPROVED** [review]

**Pull request merged**